src/PowerShellEditorServices.Protocol/LanguageServer/GetCommandsRequest.cs

```diff
@@ -0,0 +1,54 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using Microsoft.PowerShell.EditorServices.Protocol.MessageProtocol;
+using System.Collections.Generic;
+using System.Management.Automation;
+
+namespace Microsoft.PowerShell.EditorServices.Protocol.LanguageServer
+{
+    /// <summary>
+    /// Defines a class that describes the request to get all PowerShell Commands from the current session
+    /// </summary>
+    public class GetAllCommandsRequest
+    {
+        public static readonly
+            RequestType<List<PSAllCommandsMessage>, object, object, object> Type =
```

Comment:
Is there a way to avoid using `object` so much? Ideally we would be able to express the API endpoint with types

Reply:
I would love this, but I'm not sure what else I could use. It doesn't like the use of void here, and when I attach the debugger and step into the constructor this is what I see:
![image](https://user-images.githubusercontent.com/30301021/44069047-ebc134d6-9f31-11e8-89cd-d0612d7b8c8f.png)

Reply:
I might be wrong about this, but my understanding is that .NET will instantiate the generic with the type you ask it for; `TResult` is object because that's what it's told to instantiate as (i.e. .NET doesn't have the capacity to work out a better type).

What we should do is work out what the data being sent across is and if it's already described by some type, use that type, or otherwise create a new type to describe that data. That should also solve the issue below as well. I'll give the code a proper read/compile tomorrow and hopefully we'll see if there's a nice solution 😄

Reply:
Works for me. I'm just going through the vscode-powershell PR comments now and will push them up.